### PR TITLE
fix(client): FAQ question header cursor change

### PR DESCRIPTION
**Pull request by AyushDhimanIITR**
Close: #7064 

## PR Details

Changed cursor on hovering the FAQ question header to cursor pointer.

### Summary
🏷️🛠️🎨

This pull request updates several components in the `ui/design-system` package to use the new design system based on Material UI and to improve their appearance and consistency. It also adds a new `Chip` component and a new `FlexItem` component with subcomponents. Additionally, it adds a new GitHub action workflow for labeling pull requests and a new optional field for project descriptions in the `git-sync-manager` package.

> _Sing, O Muse, of the mighty pull request_
> _That brought new glory to the `ui/design-system` package_
> _And added many a component and story, well-crafted and styled_
> _With the colors and fonts of the new design system, splendid and bright._

### Walkthrough
*  Add a new workflow file for the GitHub action that labels pull requests based on the affected projects and branches ([link](https://github.com/amplication/amplication/pull/7111/files?diff=unified&w=0#diff-eadf2c6de99f6632a36af544d29eee58f6d868620bb798c56284cf43b0ee8c60R1-R93))
*  Add a new optional field `description` to the `Project` type in the `git-sync-manager` package ([link](https://github.com/amplication/amplication/pull/7111/files?diff=unified&w=0#diff-aebcd788f7feb9ebc4bbfd2b35aeee1153419a772551daa1cedb30b6735d5442R1516))
*  Migrate the `Breadcrumbs` component to the new design system based on Material UI and use the `Link` component from `react-router-dom` for navigation ([link](https://github.com/amplication/amplication/pull/7111/files?diff=unified&w=0#diff-488dd68114f9b1cf76090eaf992d2174fef0d88254e29c1a3197151823b7ae85L4-R23), [link](https://github.com/amplication/amplication/pull/7111/files?diff=unified&w=0#diff-152ddac9de23ac6fa69289ead40be09020ed44dbfabf82adf7d465d7f91dc62bR4), [link](https://github.com/amplication/amplication/pull/7111/files?diff=unified&w=0#diff-152ddac9de23ac6fa69289ead40be09020ed44dbfabf82adf7d465d7f91dc62bL12-R19), [link](https://github.com/amplication/amplication/pull/7111/files?diff=unified&w=0#diff-6a09bf58d91e5ed75481ff11c39c0e665f96965948c0e00fa67a83375c7e8617L2-R24), [link](https://github.com/amplication/amplication/pull/7111/files?diff=unified&w=0#diff-6a09bf58d91e5ed75481ff11c39c0e665f96965948c0e00fa67a83375c7e8617L19-R45))
*  Migrate the `Button` component to the new design system based on Material UI and simplify the button API ([link](https://github.com/amplication/amplication/pull/7111/files?diff=unified&w=0#diff-58cb693358ff12f3549496edf3c172f28d9b09b9ab7c17a64d6840ae5a07069eL7-R11), [link](https://github.com/amplication/amplication/pull/7111/files?diff=unified&w=0#diff-58cb693358ff12f3549496edf3c172f28d9b09b9ab7c17a64d6840ae5a07069eL21-R35), [link](https://github.com/amplication/amplication/pull/7111/files?diff=unified&w=0#diff-58cb693358ff12f3549496edf3c172f28d9b09b9ab7c17a64d6840ae5a07069eL39-R83), [link](https://github.com/amplication/amplication/pull/7111/files?diff=unified&w=0#diff-58cb693358ff12f3549496edf3c172f28d9b09b9ab7c17a64d6840ae5a07069eL143-R112), [link](https://github.com/amplication/amplication/pull/7111/files?diff=unified&w=0#diff-58cb693358ff12f3549496edf3c172f28d9b09b9ab7c17a64d6840ae5a07069eL161-R126), [link](https://github.com/amplication/amplication/pull/7111/files?diff=unified&w=0#diff-58cb693358ff12f3549496edf3c172f28d9b09b9ab7c17a64d6840ae5a07069eL187-R175), [link](https://github.com/amplication/amplication/pull/7111/files?diff=unified&w=0#diff-700b90b731ee091160b3e98dec5787f3d7dbbad82a0c47b3fd82bef12b2796f6L3-R9), [link](https://github.com/amplication/amplication/pull/7111/files?diff=unified&w=0#diff-700b90b731ee091160b3e98dec5787f3d7dbbad82a0c47b3fd82bef12b2796f6L11-R100), [link](https://github.com/amplication/amplication/pull/7111/files?diff=unified&w=0#diff-31b6b30a16de0c44099e97b1a406f5dacc7b5e791b7248ab1273d3b7021e1021L13-R18), [link](https://github.com/amplication/amplication/pull/7111/files?diff=unified&w=0#diff-31b6b30a16de0c44099e97b1a406f5dacc7b5e791b7248ab1273d3b7021e1021R39), [link](https://github.com/amplication/amplication/pull/7111/files?diff=unified&w=0#diff-31b6b30a16de0c44099e97b1a406f5dacc7b5e791b7248ab1273d3b7021e1021L50-R55), [link](https://github.com/amplication/amplication/pull/7111/files?diff=unified&w=0#diff-31b6b30a16de0c44099e97b1a406f5dacc7b5e791b7248ab1273d3b7021e1021L64-R82))
*  Create a new `Chip` component that displays a small label with a background color and a text color ([link](https://github.com/amplication/amplication/pull/7111/files?diff=unified&w=0#diff-8d7dadc62909af49854d5bcb923389084ca63a5f97e41dac5d16d59ee0efc261R1-R17), [link](https://github.com/amplication/amplication/pull/7111/files?diff=unified&w=0#diff-dd3266103dbc226316f22ddbc64b924bb0c7b4876133141dc24c8a4747166aa3R1-R115), [link](https://github.com/amplication/amplication/pull/7111/files?diff=unified&w=0#diff-e2e0f76a100aa0de3ac3df66897531ed1a6beee7d89998e75ee2f0b1f87c3af2R1-R37))
*  Update the styles for the `CircleBadge` and `CircleIcon` components to use the new design system colors and to update the background color and the icon color ([link](https://github.com/amplication/amplication/pull/7111/files?diff=unified&w=0#diff-403034124eadd01a8c5682efc04a7589868f4a4831eda8a60bdbd6d4c209abecL22-R27), [link](https://github.com/amplication/amplication/pull/7111/files?diff=unified&w=0#diff-403034124eadd01a8c5682efc04a7589868f4a4831eda8a60bdbd6d4c209abecL35-R35), [link](https://github.com/amplication/amplication/pull/7111/files?diff=unified&w=0#diff-403034124eadd01a8c5682efc04a7589868f4a4831eda8a60bdbd6d4c209abecL43-R43), [link](https://github.com/amplication/amplication/pull/7111/files?diff=unified&w=0#diff-403034124eadd01a8c5682efc04a7589868f4a4831eda8a60bdbd6d4c209abecL51-R51), [link](https://github.com/amplication/amplication/pull/7111/files?diff=unified&w=0#diff-403034124eadd01a8c5682efc04a7589868f4a4831eda8a60bdbd6d4c209abecL59-R59), [link](https://github.com/amplication/amplication/pull/7111/files?diff=unified&w=0#diff-403034124eadd01a8c5682efc04a7589868f4a4831eda8a60bdbd6d4c209abecL67-R67), [link](https://github.com/amplication/amplication/pull/7111/files?diff=unified&w=0#diff-403034124eadd01a8c5682efc04a7589868f4a4831eda8a60bdbd6d4c209abecL75-R75), [link](https://github.com/amplication/amplication/pull/7111/files?diff=unified&w=0#diff-7eab80f07efc780f8a44b5ff562189629f05f66c0341bb59cfa597beb60e5193L16-R16), [link](https://github.com/amplication/amplication/pull/7111/files?diff=unified&w=0#diff-7eab80f07efc780f8a44b5ff562189629f05f66c0341bb59cfa597beb60e5193L27-R33))
*  Update the styles for the `CodeEditor` component to use the new design system colors and to update the border color and the background color ([link](https://github.com/amplication/amplication/pull/7111/files?diff=unified&w=0#diff-4f2cf9ebd31bd845a0864736ea820dbac43c095e236306346155b9680c362228L6-R9), [link](https://github.com/amplication/amplication/pull/7111/files?diff=unified&w=0#diff-4f2cf9ebd31bd845a0864736ea820dbac43c095e236306346155b9680c362228L16-R16))
*  Update the styles for the `ConfirmationDialog` and `Dialog` components to use the new design system colors and to update the border color and the background color ([link](https://github.com/amplication/amplication/pull/7111/files?diff=unified&w=0#diff-60f73e70584672e9920c341c11d62dff8f5f0979017769ad2b898bad6283efd3L5-R5), [link](https://github.com/amplication/amplication/pull/7111/files?diff=unified&w=0#diff-60f73e70584672e9920c341c11d62dff8f5f0979017769ad2b898bad6283efd3L21-R21), [link](https://github.com/amplication/amplication/pull/7111/files?diff=unified&w=0#diff-84e9b3dcf7d307e4e43790d9faa2abd809349fc4f3db560e3fbce1cb11368d1cL9-R9), [link](https://github.com/amplication/amplication/pull/7111/files?diff=unified&w=0#diff-84e9b3dcf7d307e4e43790d9faa2abd809349fc4f3db560e3fbce1cb11368d1cL15-R15), [link](https://github.com/amplication/amplication/pull/7111/files?diff=unified&w=0#diff-84e9b3dcf7d307e4e43790d9faa2abd809349fc4f3db560e3fbce1cb11368d1cL36-R36), [link](https://github.com/amplication/amplication/pull/7111/files?diff=unified&w=0#diff-84e9b3dcf7d307e4e43790d9faa2abd809349fc4f3db560e3fbce1cb11368d1cL53-R53), [link](https://github.com/amplication/amplication/pull/7111/files?diff=unified&w=0#diff-84e9b3dcf7d307e4e43790d9faa2abd809349fc4f3db560e3fbce1cb11368d1cL61-R61))
*  Create a new `FlexItem` component that displays a flexible layout with optional gaps, margins, and alignment options ([link](https://github.com/amplication/amplication/pull/7111/files?diff=unified&w=0#diff-0526cd590c859dcfad50a685cf97d526ab0646e7c8ab7d479c45e925a459acd7R1-R67), [link](https://github.com/amplication/amplication/pull/7111/files?diff=unified&w=0#diff-0b5d0dc36f62bccca22af2762d39e16e643e944c68b2ff2debe54be9e9ff17a0R1-R18), [link](https://github.com/amplication/amplication/pull/7111/files?diff=unified&w=0#diff-267cc5e18c79f60f906384e8e0b0a2a60c1287c8a32bd34413475cb00f0f7af9R1-R148))
*  Update the `WithHeader` story for the `Form` component to use the new design system based on Material UI and to replace the `clear` button style with the `text` button style ([link](https://github.com/amplication/amplication/pull/7111/files?diff=unified&w=0#diff-ee93608725ec1070c6e30043c3467daabed111bd1bb3f56e4f157483bfb7db60L76-R76))

## PR Checklist

- [ ] Tests for the changes have been added
- [ ] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/CODE_OF_CONDUCT.md) file for detailed contributing guidelines.

### CODE REVIEW

**Comment by priii-25** [comment]
you can assign this to me?

**Comment by yuval-hazaz** [comment]
hi @AyushDhimanIITR thank you for your contribution.
For us to review this PR you must add the issue number in the description of the PR and make sure you were assigned to work on that issue

**Comment by AyushDhimanIITR** [comment]
Hi @yuval-hazaz I have added the issue number, can you please review it.

**Comment by AyushDhimanIITR** [comment]
Hi @yuval-hazaz can you review this PR

**Comment by AyushDhimanIITR** [comment]
@souravjain540 Issue has been fixed and PR also raised, do let me know my next steps

File: package.json [hunk 1]
```diff
@@ -129,6 +129,7 @@
     "node-fetch": "^2.6.7",
     "node-jose": "^2.2.0",
     "normalize-path": "^3.0.0",
+    "npm": "^9.0.0",
```
Comment (morhag90) [inline package.json:132]: @AyushDhimanIITR Is this added package related to the current fix?

**Review by morhag90 — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (AyushDhimanIITR) [reply package.json:132]: No, i'll correct it and then raise the PR

Comment (AyushDhimanIITR) [reply package.json]: Hey @morhag90 , I have done the changes, let me know if there's anything left

**Review by overbit — APPROVED** [review]

**Comment by overbit** [comment]
@AyushDhimanIITR thanks for you contribution. I've reverted your changes to the package-lock as they were unrelated.

**Review by morhag90 — APPROVED** [review]
